DittoHeartbeat/src/main/java/live.ditto.dittoheartbeat/Heartbeat.kt

```diff
@@ -0,0 +1,149 @@
+package live.ditto.dittoheartbeat
+
+import live.ditto.Ditto
+import android.os.Handler
+import android.os.Looper
+import androidx.compose.material3.Text
+import live.ditto.DittoConnectionType
+import live.ditto.DittoPeer
+import live.ditto.DittoPresenceObserver
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+data class HeartbeatConfig(
+    val interval: Long,
+    val collectionName: String,
+    val metaData: Map<String, Any>
+)
+
+data class HeartbeatInfo(
+    val id: Map<String, String>,
+    val lastUpdated: String,
+    val presence: Presence?,
+)
+
+data class Presence(
+    val totalConnections: Int,
+    val connections: List<DittoPeer>,
+)
+fun startHeartbeat(ditto: Ditto, config: HeartbeatConfig, onDataLog: (HeartbeatInfo) -> Unit): Handler {
+    val handler = Handler(Looper.getMainLooper())
+
+    val runnable = object : Runnable {
+        override fun run() {
+
+            val timestamp = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ", Locale.getDefault()).format(Date())
+            val deviceId = config.metaData["deviceId"] as? String ?: ""
+            val locationId = config.metaData["locationId"] as? String ?: ""
+
+            val info = HeartbeatInfo(
+                id = mapOf("deviceId" to deviceId, "locationId" to locationId),
+                lastUpdated = timestamp,
+                presence = observePeers(ditto),
+            )
+
+            addToCollection(info, config, ditto)
+
+            onDataLog(info)
+            handler.postDelayed(this, config.interval) // Repeat every config.interval milliseconds
+        }
+    }
+
+    handler.postDelayed(runnable, 0) // Initial delay
+
+    return handler
+}
+
+
+
+var presence: Presence? = null
+fun observePeers(ditto: Ditto): Presence? {
+    val presenceGraph = ditto.presence.graph
+    val totalConnections = presenceGraph.remotePeers.size
+    val connectionsList = presenceGraph.remotePeers
+    presence = Presence(totalConnections, connectionsList)
+
+    return presence
+}
+
+fun addToCollection(info: HeartbeatInfo, config: HeartbeatConfig, ditto: Ditto) {
+
+    val doc = mapOf(
+        "_id" to mapOf(
+            "deviceId" to config.metaData["deviceId"],
+            "locationId" to config.metaData["locationId"]
```

Comment:
peerKey?

Reply:
It looks like the peerKey changes on each call to `presence.graph`. This means that adding the `peerKey` in the `_id` will cause a new document to be created each time.

Reply:
My two cents here.

PeerKey is an array of bytes.
Do you think makes sense to have a hash representation of the bytes?
A hash is at least more readable.

Reply:
Yes it would be best to use the hash. The peerKey shouldn't change on each call that is strange

Reply:
Yep, realized my issue. I've converted it to a hash and it's working as expected